aten/src/ATen/native/xnnpack/Convolution.cpp

```diff
@@ -228,6 +151,100 @@ Tensor create_and_run(
+      "Reason: The provided (weight, bias, padding, stride, dilation, groups, output_min, output_max) "
+      "parameters are either invalid individually or their combination is not supported by XNNPACK.");
+
+  xnn_operator_t convolution_op{};
+
+  const xnn_status create_status = xnn_create_convolution2d_nhwc_f32(
+      padding_expanded[Layout::Parameter::height],                    // input_padding_top
+      padding_expanded[Layout::Parameter::width],                     // input_padding_right
+      padding_expanded[Layout::Parameter::height],                    // input_padding_bottom
+      padding_expanded[Layout::Parameter::width],                     // input_padding_left
+      weight_nhwc.size(Layout::Filter::height),                       // kernel_height
+      weight_nhwc.size(Layout::Filter::width),                        // kernel_width
+      stride_expanded[Layout::Parameter::height],                     // subsampling_height
+      stride_expanded[Layout::Parameter::width],                      // subsampling_width
+      dilation_expanded[Layout::Parameter::height],                   // dilation_height
+      dilation_expanded[Layout::Parameter::width],                    // dilation_width
+      groups,                                                         // groups
+      weight_nhwc.size(Layout::Filter::input),                        // group_input_channels
+      weight_nhwc.size(Layout::Filter::output) / groups,              // group_output_channels
+      weight_nhwc.size(Layout::Filter::input) * groups,               // input_pixel_stride
+      weight_nhwc.size(Layout::Filter::output),                       // output_pixel_stride
+      weight_nhwc.data_ptr<float>(),                                  // kernel
+      (bias && bias->defined()) ? bias->data_ptr<float>() : nullptr,  // bias
+      output_min,                                                     // output_min
+      output_max,                                                     // output_max
+      0u,                                                             // flags
+      &convolution_op);                                               // operator
+
+  TORCH_CHECK(
+      xnn_status_success == create_status,
+      "xnn_create_convolution2d_nhwc_f32 failed!");
+
+  return ContextConv2D(
+      Operator(convolution_op),
+      weight_nhwc.sizes().vec(),
+      padding_expanded,
+      stride_expanded,
+      dilation_expanded
+  );
+}
+
+c10::intrusive_ptr<xnnpack::XNNPackConv2dOpContext> Conv2dPrePack::operator()(
+  Tensor weight,
+  c10::optional<Tensor> bias,
+  std::vector<int64_t> stride,
+  std::vector<int64_t> padding,
+  std::vector<int64_t> dilation,
+  int64_t groups
+  ) {
+    return xnnpack::XNNPackConv2dOpContext::create_context(
+        std::move(weight),
+        std::move(bias),
+        std::move(padding),
+        std::move(stride),
+        std::move(dilation),
+        groups,
+        {},
+        {});
```

Comment:
Do we need min/max here to fuse relu?

Reply:
So my plan was to make the necessary changes, including ones needed in op API, in the PR for relu fusion.